### ISSUE TITLE
Synology Migration -> Stuck in Duplicate Node Key Loop

### ISSUE THREAD

**Issue by AyrtonB** [tailscale/tailscale#12801]
### What is the issue?

Issue
* Duplicated one Synology (that had Tailscale installed) to another using their "Migration Assistant" tool
* Tailscale was automatically set up on the new Synology but in my console both Synologys were showing up under the same Tailscale device with a "Duplicate Node Keys" error

<br>

Attempt to Solve
* I uninstalled and reinstalled Tailscale on both devices from the Synology package center
* I also removed the device from the Tailscale console
* I reinstalled Tailscale on both devices
* I used `sudo tailscale up --force-reauth` on both devices, the URL that I followed then showed different node keys for each device
* I go back to the console and can only see the device that was authenticated last
* Rerunning `sudo tailscale up --force-reauth` on the device that wasn't showing up in the console makes it appear again, but removes the other device
* Everytime I run `sudo tailscale up --force-reauth` it now switches which device is connected to the console

<br>

How can I reset this properly so that I can connect both devices to Tailscale?

### Steps to reproduce

Duplicated one Synology (that had Tailscale installed) to another using their "Migration Assistant" tool

### Are there any recent changes that introduced the issue?

Duplicated one Synology (that had Tailscale installed) to another using their "Migration Assistant" tool

### OS

Synology

### OS version

DSM 7

### Tailscale version

1.58.2

### Bug report

BUG-48c9bbdcaa42202a8bb2f3e309f9270470f945ce4b75c4e5df36ea15df50d833-20240713111321Z-46722db7165110b3

**Comment by AyrtonB** [comment]
This person has had the same issue (also unsolved) - https://forum.tailscale.com/t/synology-nas-after-migration-to-new-nas/4646/1

**Comment by stevetrease** [comment]
Have you tried completely uninstalling Tailscale on one of the Synologies and the reinstalling and deauthorising?

[https://tailscale.com/kb/1069/uninstall?tab=synology](url)

**Comment by AyrtonB** [comment]
Yes Tailscale was uninstalled and then reinstalled on both devices from the Synology package center

**Comment by stevetrease** [comment]
Did you also manually remove the files in /var/packages/Tailscale as instructed in the Tailscale article?

```
Uninstall Tailscale from Synology package center.
Delete Synology node from Tailscale admin console.
Remove the files at the following paths:

/var/packages/Tailscale/var/*
/var/packages/Tailscale/etc/*
```

**Comment by goshsakesalive** [comment]
Also experiencing this.  Used Migration Assistant to load my newer DS923+ from my older DS918+.  Deleted the Tailscale package from both systems.  SSH sessions showed /var/packages/Tailscale absent from both so there was no way to remove files from that path.  Loaded Tailscale package 1.58.2-700058002 on DS923+, reauthorized, and it showed up in the Machines list.  Loaded the same Tailscale package on the DS918+ and reauthorized.  The Machines list now showed the "Machine name" as the ds923 but the "OS hostname" was updated to that of the ds918.  The "ID" was unchanged after the second reauthorization, but the "Node key" was updated to a new value.

**Comment by goshsakesalive** [comment]
OK, here it is-- stop the package, then SSH into your system and delete **/volume1/@appdata/Tailscale/tailscaled.state**, then run the package again.  That file gets copied by the Migration Assistant and it holds a "machinekey" that needs to be regenerated for the system to rejoin the tailnet as a unique entity.

**Comment by once375ml** [comment]
> OK, here it is-- stop the package, then SSH into your system and delete **/volume1/@appdata/Tailscale/tailscaled.state**, then run the package again. That file gets copied by the Migration Assistant and it holds a "machinekey" that needs to be regenerated for the system to rejoin the tailnet as a unique entity.

You are my lifesaver. You solve my problem! Thanks.
